src/shared/transport.ts

```diff
@@ -39,8 +40,11 @@ export interface Transport {
 
   /**
    * Callback for when a message (request or response) is received over the connection.
+   * 
+   * Includes the authInfo if the transport is authenticated.
+   * 
    */
-  onmessage?: (message: JSONRPCMessage) => void;
+  onmessage?: (message: JSONRPCMessage, authInfo?: AuthInfo) => void;
```

Comment:
Please wrap this in an object, like this:
```suggestion
  onmessage?: (message: JSONRPCMessage, extra?: { authInfo?: AuthInfo }) => void;
```

This is better for extensibility later, and gives parameters clear names for destructuring.